docs/DeveloperGuide.adoc

```diff
@@ -142,71 +155,356 @@ image::StorageClassDiagram.png[]
+class that that would need to change if implementation of any of the `Type` changes, our implementation ensures that
+our many command classes would only have a single reason to change. Moreover, our current implementation also
+reduces double work as `Parser` will not have to parse `Type` in the user input to create the `ABCCommand`, then only to
+be switch-cased again in `ABCCommand`.
+****
+
+=== Me Feature
+This feature allows a user to make changes and updates his/her user profile which is reflected by the user profile panel.
+At the same time, it also allows user to personalise his/her experience using the app with certain sub-features. They are
+elaborated below:
+
+* Edit user profile image
+
+* Edit user main data
+
+* Input, store and generate motivational quotes
+
+* Change background and font color
+
+* Set reminders or take notes
+
+==== Edit User Profile Image
+This allows the specific user to input and update his/her user profile picture.
+
+==== Edit User Main Data
+This allows one specific user to modify and update user information that includes `name`, `phone`, `email1`, `github`,
+`university`, `major`, `from`, `to`, `cap`.
+
+Below is an example usage scenario and how the edit user data command works.
+
+* User launches the application for the first time. The User Profile and Data will be initialized with the initial json
+data stored.
+
+image::user-default-data.png[]
+
+* User executes `me n/NAME p/PHONE e/EMAIL g/GITHUB u/UNIVERSITY m/MAJOR f/FROM t/TO c/CAP` so as to update the Person
+object currently stored in Model as well as Storage.
+
+ me n/Nham Hung p/91608840 e/nhamhung.gttn@gmail.com g/nhamhung u/NUS m/CS f/10-2020 t/11-2020 c/5.0
+
+* The Ui User Data Table will be updated accordingly.
+
+image::user-data.png[]
+
+Command sequence:
+
+1. User type `me` command in the command box.
+2. Command is executed by Logic Manager.
+3. Command is parsed by `ResumeBookParser` which identifies what type of command it is. An `EditUserParser` is returned accordingly.
+4. `EditUserParser` extracts out different components of the command into an `EditUserDescriptor` object and returns an `EditUserCommand`
+with the `EditUserDescriptor` object which contains information on which attributes of user data is updated or kept unchanged.
+5. `EditUserCommand` then calls `execute()` which first gets the existing user in Model as `userToEdit`. Then it creates a new
+`editedUser` based on `EditUserDescriptor`. It will then update the current `userToEdit` in Model with `editedUser`. Afterwards,
+a `CommandResult` is returned to Logic with data and feedback to be displayed to the user.
+6. Feedback acknowledge is displayed by `ResultDisplay`. User profile changes are displayed by JavaFx `TableView`.
+
+The following sequence diagram shows how the `me` feature allows user to edit the user profile:
+
+image::MeSequenceDiagram.png[]
```

Comment:
Me Feature Implementation Sequence Diagram is wrong → control at EditUserCommand has two concurrent calls → getUser() and new Person("Quoc Hung") called before either returns anything

Reply:
I agree with @ChesterSim 's comment.